### PR TITLE
kie-kogito-serverless-operator-361: Add data-index and job service st…

### PR DESCRIPTION
**Pull request by wmedvede**
…artupProbes to the workflow Deployment

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [X] Add or Modify a unit test for your change
- [X] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: controllers/platform/services/services.go [hunk 1]
```diff
@@ -295,7 +296,9 @@ func (j JobServiceHandler) GenerateServiceProperties() (*properties.Properties,
 
 func (j JobServiceHandler) GenerateWorkflowProperties() (*properties.Properties, error) {
 	props := properties.NewProperties()
-	props.Set(constants.JobServiceRequestEventsURL, fmt.Sprintf("%s/v2/jobs/events", generateServiceURL(constants.KogitoProcessEventsProtocol, j.platform.Namespace, j.GetServiceName())))
+	jobServiceUrl := generateServiceURL(constants.KogitoProcessEventsProtocol, j.platform.Namespace, j.GetServiceName())
+	props.Set(constants.KogitoJobServiceURL, jobServiceUrl)
+	props.Set(constants.JobServiceRequestEventsURL, fmt.Sprintf("%s/v2/jobs/events", jobServiceUrl))
```
Comment (ricardozanini) [inline controllers/platform/services/services.go:301]: Is this part of this PR? Isn't only the health check endpoint needed?

**Review by ricardozanini — COMMENTED** [review]
Thank you! But I'm missing the `startupProbe` in the workflow container. 😅

**Comment by wmedvede** [comment]
> Thank you! But I'm missing the `startupProbe` in the workflow container. 😅

this startupProbe is already being set here: https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/7ba75e5c8573829199f4d85f2a6f5e86a1fc597e/controllers/profiles/common/object_creators.go#L142

**Comment by ricardozanini** [comment]
@wmedvede can u rebase?

**Review by ricardozanini — APPROVED** [review]

**Review by jakubschwan — APPROVED** [review]

**Pull request merged**